src/core/qgstaskmanager.cpp

```diff
@@ -142,6 +142,27 @@ QList<QgsMapLayer *> QgsTask::dependentLayers() const
   return _qgis_listQPointerToRaw( mDependentLayers );
 }
 
+bool QgsTask::waitForFinished( int timeout )
```

Comment:
@nyalldawson 

I merged this as is. But I'd be happy to continue this discussion here.